lib/smb.c

```diff
@@ -915,7 +915,7 @@ static CURLcode smb_connection_state(struct Curl_easy *data, bool *done)
  */
 static void get_posix_time(time_t *out, curl_off_t timestamp)
 {
-  timestamp -= 116444736000000000;
+  timestamp -= 116444736000000000LL;
```

Comment:
```suggestion
  timestamp -= CURL_OFF_T_C(116444736000000000);
```
The size of `curl_off_t` depends on platform. Make sure
to use the matching suffix by using this macro.

Reply:
Updated as suggested. Compiles without warning on target.